drivers/slipdev/slipdev.c

```diff
@@ -207,6 +208,11 @@ static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
                 return -ENOBUFS;
             }
         } while (byte != SLIPDEV_END);
+
+        if (++dev->rx_done != dev->rx_queued) {
+            DEBUG("slipdev: pkt still in queue");
+            netdev->event_callback(netdev, NETDEV_EVENT_RX_COMPLETE);
```

Comment:
Note that we actually should make use of the event use in `gnrc_netif` by also posting events in case of `NETDEV_EVENT_RX_COMPLETE` and the others. This is especially useful when a driver can tell from within ISR that a frame was received / TX completed / ... This is easily possible for peripheral network devices, but also for SPI attached devices with multiple status pins.

This would have the positive side affect to avoid the recursion. But it does make sense to do so before merging this.

Reply:
> Note that we actually should make use of the event use in gnrc_netif by also posting events in case of NETDEV_EVENT_RX_COMPLETE and the others. This is especially useful when a driver can tell from within ISR that a frame was received / TX completed / ... This is easily possible for peripheral network devices, but also for SPI attached devices with multiple status pins.

Strong +1.